nutree/typed_tree.py

```diff
@@ -224,6 +224,7 @@ def add_child(
         deep: bool = None,
         data_id=None,
         node_id=None,
+        factory: "Node" = None,
```

Comment:
Should this be `: Type[Node]` or similar, since we expect a class instead of an instance?